vignettes/dstidyverse.Rmd.orig

```diff
@@ -26,111 +26,134 @@ options(knitr.duplicate.label = "allow")
-        source.times = "c",
-        source.length.out = NULL,
-        source.each = "c",
-        x1.includes.characters = FALSE,
-        newobj = "ONES")
-
-ds.dataFrameSubset(df.name = "mtcars",
-                   V1.name = "ONES",
-                   V2.name = "ONES",
-                   Boolean.operator = "==",
-                   keep.cols = c(1, 3, 6),
-                   rm.cols = NULL,
-                   keep.NAs = FALSE,
-                   newobj = "test")
-
-ds.colnames("test")
+ds.select(df.name = "mtcars", tidy_select = list(mpg, disp, wt), newobj = "new_df")
+ds.colnames("new_df")
 ```
-## ds.select
-ds.select implements all the flexibility of dplyr::select in subsetting columns. This includes:
-(i) ability to pass enquoted column names which are evaluated as columns of .data
-(ii) ability to use tidyselect helper functions, such as `contains`.
-
-For example:
-
-### Subset columns by name
+#### Specify columns to exclude
 ```{r}
 ds.colnames("mtcars")
-ds.select(.data = "mtcars", tidy_select = list(mpg, disp, wt), newobj = "test")
-ds.colnames("test")
+ds.select(df.name = "mtcars", tidy_select = list(!mpg), newobj = "new_df")
+ds.colnames("new_df")
 ```
-
-### Specify column to exclude
+#### Specify a range of columns to keep
 ```{r}
 ds.colnames("mtcars")
-ds.select(.data = "mtcars", tidy_select = list(!mpg), newobj = "test")
-ds.colnames("test")
+ds.select(df.name = "mtcars", tidy_select = list(mpg:drat), newobj = "new_df")
+ds.colnames("new_df")
 ```
-### Specify range of columns
+#### Specify column names using a regular expression
 ```{r}
 ds.colnames("mtcars")
-ds.select(.data = "mtcars", tidy_select = list(mpg:drat), newobj = "test")
-ds.colnames("test")
+ds.select(df.name = "mtcars", tidy_select = list(matches('[aeiou]')), newobj = "new_df")
+ds.colnames("new_df")
 ```
-### Match by first letter
+#### Combine multiple conditions with '&'
 ```{r}
 ds.colnames("mtcars")
-ds.select(.data = "mtcars", tidy_select = list(starts_with('m')), newobj = "test")
-ds.colnames("test")
+ds.select(df.name = "mtcars", tidy_select = list(starts_with('c') & ends_with('b')), newobj = "new_df")
+ds.colnames("new_df")
 ```
-### Match by last letter
+#### Combine multiple conditions with '|'
 ```{r}
 ds.colnames("mtcars")
-ds.select(.data = "mtcars", tidy_select = list(ends_with('m')), newobj = "test")
-ds.colnames("test")
+ds.select(df.name = "mtcars", tidy_select = list(starts_with('c') | ends_with('b')), newobj = "new_df")
+ds.colnames("new_df")
 ```
-### Pattern matching
+### Rename
+`ds.rename` renames columns within a serverside dataframe. Column names are passed unquoted and are
+evaluated as column names within `df.name`. Below are some examples, see `dplyr::rename` for full details.
+
 ```{r}
 ds.colnames("mtcars")
-ds.select(.data = "mtcars", tidy_select = list(matches('[aeiou]')), newobj = "test")
-ds.colnames("test")
+ds.rename(df.name = "mtcars", tidy_select = list(effiency = mpg, power = hp), newobj = "new_df")
```

Comment:
#> Error in str_sub(str_replace_all(string, pattern = fixed("list("), replacement = ""), : could not find function "str_sub"

Reply:
Could you check for me what version of the `stringr` package you have installed? I might need to specify a minimum.

Reply:
stringr 1.5.1